src/app/auth/register/register.component.ts

```diff
@@ -92,23 +73,30 @@ export class RegisterComponent implements OnInit, DoCheck {
 
     const { email, displayName, institution, website, password } = this.registerForm.controls;
 
+    let userInput: UserCreateInput;
+    userInput = {
+      'email' : email?.value,
+      'displayName' : displayName?.value,
+      'passwordHash':''
+    }
+    if (true /*website != ''*/){userInput.website = website?.value;}
+    if (true /*institution != ''*/){userInput.institution = institution?.value;}
+
     this.$http.post('/api/auth/register',
     {
-      email:email?.value.toLowerCase(),
-      displayName:displayName?.value,
-      institution:institution?.value,
-      website:website?.value,
-      password:password?.value
+      userInput: userInput,
+      password: password?.value
     }, { observe: 'response' }).subscribe((response: HttpResponse<Object>) => {
 
       if(response.ok && (response.body?.['error'] === undefined)){
         //User Created
       }
     },
     err => {
-      if (err.error?.['error'].message === UniqueError.E_ERR){this.uniqueEmailFlag = false}
-      if (err.error?.['error'].message === UniqueError.D_ERR){this.uniqueDisplayNameFlag = false}
-      if (err.error?.['error'].message === UniqueError.B_ERR){this.uniqueEmailFlag = false;this.uniqueDisplayNameFlag = false}
+      console.log(err)
+      if (err.status === 403){
+        this.registerForm.get('formStatus')?.setValue(err.error.message)
```

Comment:
This is not the correct way to display an error message.

Set another property on the class to handle error messages.